### ISSUE TITLE
Discarded constructor parameter produces diagnostic after converting to primary constructor

### ISSUE THREAD

**Issue by drewnoakes** [dotnet/roslyn#68869]
**Version Used**: Version 17.8.0 Preview 1.0 [33902.16.main]

**Steps to Reproduce**:

```c#
class C
{
    public C(object _) { }
}
```

Use codefix to convert to primary constructor:

```c#
class C(object _)
//             ^ CS9113 Parameter '_' is unused
{
}
```

In dotnet/project-system and CPS we often need to specify a discarded constructor parameter in this way to enforce scoping for MEF components. When investigating dogfooding primary constructors, I hit this difference. Note that the original code doesn't produce any diagnostics.

**Diagnostic Id**:

CS9113

**Expected Behavior**:

Discarded parameter does not produce a diagnostic.

**Actual Behavior**:

CS9113 is produced.

**Comment by mavasani** [comment]
Moving to Area-Compilers as this is about a redundant compiler warning.

**Comment by AlekseyTs** [comment]
>**Expected Behavior**:

>Discarded parameter does not produce a diagnostic.

As far as I know, language doesn't have a concept of a discarded parameter.  The behavior is by design and is consistent with behavior in records:
```
record C1(object _) //  warning CS8907: Parameter '_' is unread. Did you forget to use it to initialize the property with that name?
{
    public object _;
}
```

**Comment by MaceWindu** [comment]
Another case: asp.net core middleware constructors [required](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/middleware/write?view=aspnetcore-8.0#middleware-class) to have parameter of RequestDelegate type even if it is not used and user need to suppress CS9113 if primary constructor syntax used.
